src/platforms/windows/daemon/wireguardutilswindows.cpp

```diff
@@ -566,10 +584,23 @@ bool WireguardUtilsWindows::deleteRoutePrefix(const IPAddress& prefix) {
   return result == NO_ERROR;
 }
 
-bool WireguardUtilsWindows::addExclusionRoute(const IPAddress& prefix) {
-  return m_routeMonitor.addExclusionRoute(prefix);
-}
+bool WireguardUtilsWindows::excludeLocalNetworks(
+    const QList<IPAddress>& addresses) {
+  // If the interface isn't up then something went horribly wrong.
+  Q_ASSERT(m_routeMonitor);
+
+  // For each destination - attempt to exclude it from the VPN tunnel.
+  bool result = true;
+  for (const IPAddress& prefix : addresses) {
+    if (!m_routeMonitor->addExclusionRoute(prefix)) {
+      result = false;
+    }
+  }
+
+  // Permit LAN traffic through the firewall.
+  if (!WindowsFirewall::instance()->enableLanBypass(addresses)) {
+    result = false;
```

Comment:
Should we be removing the exclusion route here, given if we end up here that route exist but might be blocked by the firewall?